app/main.py

```diff
@@ -1,3 +1,53 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | str) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        elif isinstance(other, (int, float)):
+            return Distance(
+                km=self.km + other
+            )
+
+    def __iadd__(self, other: int | str) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: int | str) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(
+                km=self.km * other
+            )
+
+    def __truediv__(self, other: int | str) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(
+                km=round(self.km / other, 2)
+            )
+
+    def __lt__(self, other: int | float) -> bool:
```

Comment:
The comparison methods (`__lt__`, `__gt__`, `__eq__`, `__le__`, `__ge__`) should also support comparisons with `Distance` objects, not just numeric types, as per the task requirements .